cmd/dep/init_test.go

```diff
@@ -33,3 +37,44 @@ func TestIsStdLib(t *testing.T) {
 		}
 	}
 }
+
+func TestGetProjectPropertiesFromVersion(t *testing.T) {
+	cases := []struct {
+		version  gps.Version
+		expected gps.Version
+	}{
+		{
+			version:  gps.NewBranch("foo-branch").Is("some-revision"),
+			expected: gps.NewBranch("foo-branch"),
+		},
+		{
+			version:  gps.NewVersion("foo-version").Is("some-revision"),
+			expected: gps.NewVersion("foo-version"),
+		},
+		{
+			version:  gps.Revision("alsjd934"),
+			expected: nil,
+		},
+		// This fails. Hence, testing separately below.
+		// {
+		// 	version: gps.NewVersion("v1.0.0"),
+		// 	expected: gps.NewVersion("^1.0.0"),
+		// },
+	}
+
+	for _, c := range cases {
+		actualProp := getProjectPropertiesFromVersion(c.version)
+		if c.expected != actualProp.Constraint {
+			t.Fatalf("Expected %q to be equal to %q", actualProp.Constraint, c.expected)
+		}
+	}
+
+	outsemver := getProjectPropertiesFromVersion(gps.NewVersion("v1.0.0"))
+	expectedSemver, _ := gps.NewSemverConstraint("^1.0.0")
+	// Comparing outsemver.Constraint and expectedSemver fails with error
+	// "comparing uncomparable type semver.rangeConstraint", although they have
+	// same value and same type "gps.semverConstraint" as per "reflect".
+	if outsemver.Constraint.String() != expectedSemver.String() {
+		t.Fatalf("Expected %q to be equal to %q", outsemver, expectedSemver)
+	}
+}
```

Comment:
Yeah, the issue here is that `semver.rangeConstraint` contains a slice, which means [it's not directly comparable](https://golang.org/ref/spec#Comparison_operators), even though that slice is almost always empty, including in this case.

The simple solution here is to use `reflect.DeepEqual()` to compare them instead.

Apart from that, I think the tests are actually fine as-is; they focus in on the key issue, which is stripping out the underlying revision, and making sure that we default to the caret. The only ones I might add, just for completeness, is a duplicate set where there isn't a revision attached (the `Version` is actually a `gps.UnpairedVersion`). This case shouldn't occur in the wild, but it's to test for and we should know if we ever regress in handling it, for some reason.